app/experimenter/nimbus-ui/src/components/PageHome/FilterBar/index.tsx

```diff
@@ -74,6 +82,71 @@ type FilterSelectProps<
   optionLabelName: keyof NonNullFilterOption<K>;
 };
 
+const GroupedFilterSelect = <
+    K extends FilterValueKeys,
+    T extends NonNullFilterOptions<K>,
+    >({
+        filterValue,
+        onChange,
+        filterValueName,
+        fieldLabel,
+        fieldOptions,
+        optionLabelName,
+      }: FilterSelectProps<K, T>) => {
+  const filterOptions = useMemo(
+      () =>
+          (fieldOptions! as NonNullFilterOptions<K>).filter(
+              (option): option is NonNullable<typeof option> => !!option,
+          ),
+      [fieldOptions],
+  );
+
+  const featureOptions: any[] = [];
```

Comment:
I am not familiar with typescript and I had trouble correctly typing the elem. @jaredlockhart could I get some pointers to the right direction?